bson/json_util.py

```diff
@@ -946,3 +961,87 @@ def default(obj: Any, json_options: JSONOptions = DEFAULT_JSON_OPTIONS) -> Any:
             # original value, when float() is called on it.
             return {"$numberDouble": str(repr(obj))}
     raise TypeError("%r is not JSON serializable" % obj)
+
+
+def get_size(obj: Any, max_size: int, current_size: int = 0) -> int:
+    """Recursively finds size of objects"""
+    if current_size >= max_size:
+        return current_size
+    if isinstance(obj, ObjectId):
```

Comment:
This seems like another good place to use a lut based on object type, and store the lengths of these strings as constants.